components/calendar/src/any_calendar.rs

```diff
@@ -344,6 +354,49 @@ impl AnyCalendar {
             Self::Iso(_) => "Iso",
         }
     }
+
+    pub fn kind(&self) -> AnyCalendarKind {
+        match *self {
+            Self::Gregorian(_) => AnyCalendarKind::Gregorian,
+            Self::Buddhist(_) => AnyCalendarKind::Buddhist,
+            Self::Japanese(_) => AnyCalendarKind::Japanese,
+            #[allow(clippy::expect_used)] // Invariant known at compile time
+            Self::Ethiopic(ref e) => e
+                .any_calendar_kind()
+                .expect("Ethiopic calendar known to have an AnyCalendarKind"),
+            Self::Indian(_) => AnyCalendarKind::Indian,
+            Self::Coptic(_) => AnyCalendarKind::Coptic,
+            Self::Iso(_) => AnyCalendarKind::Iso,
+        }
+    }
+
+    /// Given an AnyCalendar date, convert that date to another AnyCalendar date in this calendar,
+    /// if conversion is needed
+    pub fn convert_any_date<'a>(
+        &'a self,
+        date: &Date<impl AsCalendar<Calendar = AnyCalendar>>,
+    ) -> Date<Ref<'a, AnyCalendar>> {
```

Comment:
Discussion (here and below): This function makes sense but I'm not sure if I like the signature or whether we need it at all, since it's just candy over an operation that you can do with other public APIs.

Reply:
Yeah, but the code is awkward and I found myself needing it a lot before I scaled back the scope of this PR to not be doing conversions. If we do end up doing conversions we'll probably need this. If we _don't_, we'll want to expose this to clients since I consider `any_calendar_kind()` to be a largely _internal_ API.